atomic_reactor/plugins/pre_resolve_module_compose.py

```diff
@@ -206,13 +216,14 @@ def _resolve_compose(self):
         # makes things simpler.
         pdc_client = PDCClient(server=self.pdc_url, ssl_verify=not self.pdc_insecure, develop=True)
 
-        if self.compose_id is None:
-            if self.module_version is not None:
-                source = self.module_name + '-' + self.module_stream + '-' + self.module_version
-            else:
-                source = self.module_name + '-' + self.module_stream
+        if self.module_version is not None:
+            source_spec = self.module_name + '-' + self.module_stream + '-' + self.module_version
```

Comment:
What about:
```
source_spec = "{0}-{1}".format(self.module_name, self.module_stream)
if self.module_version:
    source_spec = "{0}-{1}".format(source_spec, self.module_version)
```

This would avoid concatenation and convert into strings properly